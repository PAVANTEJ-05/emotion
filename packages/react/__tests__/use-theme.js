/** @jsx jsx */
import 'test-utils/setup-env'
import { render } from '@testing-library/react'
import { jsx, useTheme, ThemeProvider } from '@emotion/react'

test('useTheme works', () => {
  function TestComponent(props) {
    const theme = useTheme()
    return (
      <div
        css={{ color: theme.lightGreen, '&:hover': { color: theme.darkGreen } }}
      >
        Should be green
      </div>
    )
  }

  expect(
    render(
      <ThemeProvider theme={{ lightGreen: 'green', darkGreen: 'darkgreen' }}>
        <TestComponent />
      </ThemeProvider>
    ).container.firstChild
  ).toMatchSnapshot()
})

test('Nested useTheme works', async () => {
  function TestComponent1(props) {
    const theme = useTheme()
    return (
      <div
        css={{ color: theme.lightGreen, '&:hover': { color: theme.darkGreen } }}
        {...props}
      />
    )
  }

  function NestedComponent(props) {
    const theme = useTheme()
    return (
      <div
        css={{
          color: theme.lightGreen,
          '&:hover': { color: theme.darkGreen }
        }}
        {...props}
      />
    )
  }

  function TestComponent2(props) {
    return (
      <TestComponent1>
        Should be green
        <ThemeProvider
          theme={{ lightGreen: 'lawngreen', darkGreen: 'seagreen' }}
        >
          <NestedComponent>Should be lawngreen</NestedComponent>
        </ThemeProvider>
      </TestComponent1>
    )
  }

  expect(
    render(
      <ThemeProvider theme={{ lightGreen: 'green', darkGreen: 'darkgreen' }}>
        <TestComponent2 />
      </ThemeProvider>
    ).container.firstChild
  ).toMatchSnapshot()
})
