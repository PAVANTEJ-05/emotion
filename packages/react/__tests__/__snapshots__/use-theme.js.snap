// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nested useTheme works 1`] = `
.emotion-0 {
  color: green;
}

.emotion-0:hover {
  color: darkgreen;
}

.emotion-1 {
  color: lawngreen;
}

.emotion-1:hover {
  color: seagreen;
}

<div
  class="emotion-0"
>
  Should be green
  <div
    class="emotion-1"
  >
    Should be lawngreen
  </div>
</div>
`;

exports[`useTheme works 1`] = `
.emotion-0 {
  color: green;
}

.emotion-0:hover {
  color: darkgreen;
}

<div
  class="emotion-0"
>
  Should be green
</div>
`;
